nav2_mppi_controller/src/critics/cost_critic.cpp

```diff
@@ -153,27 +171,36 @@ void CostCritic::score(CriticData & data)
       // The footprintCostAtPose will always return "INSCRIBED" if footprint is over it
       // So the center point has more information than the footprint
       if (!worldToMapFloat(Tx, Ty, x_i, y_i)) {
-        if (!is_tracking_unknown_) {
-          traj_cost = collision_cost_;
-          trajectory_collide = true;
-          break;
-        }
-        pose_cost = 255.0f;  // NO_INFORMATION in float
+        pose_cost = nav2_costmap_2d::NO_INFORMATION;
       } else {
+        // Get center cost
         pose_cost = static_cast<float>(costmap->getCost(getIndex(x_i, y_i)));
-        if (pose_cost < 1.0f) {
-          continue;  // In free space
-        }
-        if (inCollision(pose_cost, Tx, Ty, traj_yaw(i, j))) {
-          traj_cost = collision_cost_;
+        // Skip footprint cost if point cost has already exceeded inscribed obstacle cost
+        if (pose_cost >= nav2_costmap_2d::INSCRIBED_INFLATED_OBSTACLE) {
           trajectory_collide = true;
           break;
         }
+        // Get footprint cost
+        if (consider_footprint_ &&
+          (pose_cost >= possible_collision_cost_ || possible_collision_cost_ < 1.0f))
+        {
+          pose_cost = static_cast<float>(collision_checker_.footprintCostAtPose(
```

Comment:
We dont want to use the footprint cost for the cost to apply, only the collision checking process itself